components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.spec.ts

```diff
@@ -48,15 +49,119 @@ describe('ResourceDropdownComponent', () => {
     });
 
     it('displays a list of checkbox options', () => {
-      const options = Array.from(fixture.nativeElement.querySelectorAll('chef-checkbox'));
+      const options: HTMLInputElement[] = Array.from(fixture.nativeElement.querySelectorAll('chef-checkbox'));
       expect(options.length).toEqual(component.filteredResources.length);
-      options.forEach((option: HTMLInputElement, index: number) => {
+      options.forEach((option, index) => {
         const { name, checked } = component.filteredResources[index];
         expect(option.textContent).toEqual(name);
         expect(option.checked).toEqual(checked);
       });
     });
 
+    it('allows customizing "none" designator', () => {
+      component.noneSelectedLabel = 'zilch';
+      component.ngOnChanges(
+        { resources: new SimpleChange([], [], true) });
+      expect(component.label).toEqual('zilch');
+    });
+
+    using([
+      ['single unchecked resource',
+        [
+          genResource('name1', false)
+        ]],
+      ['two resources but none checked',
+        [
+          genResource('name1', false),
+          genResource('name2', false)
+        ]]
+    ], function (description: string, resources: ResourceChecked[]) {
+      it(`displays 'none' designator for ${description}`, () => {
+        component.resources = resources;
+        component.ngOnChanges(
+          { resources: new SimpleChange([], resources, true) });
+        expect(component.label).toEqual('None');
+      });
+    });
+
+    using([
+      ['single checked resource',
+        [
+          genResource('target resource', true)
+        ]],
+      ['two resources but just one checked',
+        [
+          genResource('other resource2', false),
+          genResource('target resource', true)
+        ]],
+      ['multiple resources but just one checked in the middle',
+        [
+          genResource('other resource1', false),
+          genResource('other resource2', false),
+          genResource('target resource', true),
+          genResource('other resource3', false)
+        ]],
+      ['multiple resources but just one checked at the top',
+        [
+          genResource('target resource', true),
+          genResource('other resource1', false),
+          genResource('other resource2', false),
+          genResource('other resource3', false),
+          genResource('other resource4', false)
+        ]],
+      ['multiple resources but just one checked at the bottom',
+        [
+          genResource('other resource1', false),
+          genResource('other resource2', false),
+          genResource('target resource', true)
+        ]]
+    ], function (description: string, resources: ResourceChecked[]) {
+      it(`displays resource name for ${description}`, () => {
+        component.resources = resources;
+        component.ngOnChanges(
+          { resources: new SimpleChange([], resources, true) });
```

Comment:
Oddly setting a component's `@Input` programmatically does not trigger an `OnChanges` event so we need to do that manually for the test. (`OnChanges` is triggered only when the value changes in the component's view.)